### DISCUSSION TITLE
Can we utilise Fabric connections in our workload?

### DISCUSSION THREAD

**Discussion by willwainwright** [microsoft/Microsoft-Fabric-workload-development-sample#17] [Q&A] [answered]
We are integrating our web app with a Fabric workload. We currently have a similar connections object as Microsoft Fabric. Rather than requesting users to enter into our app, is there a way our workload (with the correct granted permissions), read and access the connection strings of Microsoft Fabric connections?

**Comment by anitamayorov** [comment]
Hi @willwainwright - can you please clarify your needs and usage? Do you mean connectors to different data stores?

**Reply by willwainwright** [reply]
Hi @anitamayorov ,

We have a web app where we connect to client source data stores. With the migration to MS Fabric workload, we would like to utilise the connections that users enter into MS Fabric.
E.g. connect to a azure data lake or azure sql database.
Ideally Fabric could send a token to the workload backend with authorisation to connect to the back end.

Thanks

**Reply by anitamayorov** [reply, marked as answer]
Hi @willwainwright,

We do have examples in our Sample for such cases, for example the ```GetTablesAsync``` call in https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Backend/src/Controllers/LakeHouseController.cs
You first use workloadClientAPI to get the accessToken, then sending this to your workload backend (https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Frontend/src/controller/LakehouseExplorerController.ts#L7)
and then by using OBO with the required scopes, you can issue a request to Fabric's REST API, in our Sample those are OneLake APIs. 
So in that case the user token gets to workload backend and from there to all APIs needed.

We are working on a sample for a Sql endpoint - will update the Sample once completed. 

Did I understand your question correctly?

**Reply by pavanramineni** [reply]
@anitamayorov , did you get any chance to include the samples for SQL Endpoint ?

**Reply by anitamayorov** [reply]
Hi @pavanramineni, 
I saw you asked a similar question in this discussion https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/discussions/62 - does this answer your question?
